diff --git a/drupal/sites/all/modules/contrib/better_exposed_filters/better_exposed_filters.theme b/drupal/sites/all/modules/contrib/better_exposed_filters/better_exposed_filters.theme
index d4d3762..db688e2 100644
--- a/drupal/sites/all/modules/contrib/better_exposed_filters/better_exposed_filters.theme
+++ b/drupal/sites/all/modules/contrib/better_exposed_filters/better_exposed_filters.theme
@@ -573,6 +573,18 @@ function theme_select_as_links($vars) {
   );
 
   $output = '<div class="bef-select-as-links">';
+
+  if (isset($element['#bef_enable_fastlivefilter']) && $element['#bef_enable_fastlivefilter']) {
+    $filter_id = $element['#id'];
+    //add required js
+    drupal_add_js('
+      jQuery(function() {
+        jQuery("#fastlivefilter-'.$filter_id .'").fastLiveFilter(".search-list-'. $filter_id .' .form-item");
+        })', 'inline');
+    $output .= '<input class="search" id="fastlivefilter-'. $filter_id .'" placeholder="'. $element['#fastlivefilte_placeholder'] .'"  />';
+    $output .= '<div class="search-list search-list-'.$filter_id.'">';
+  }
+
   $output .= theme('form_element', array('element' => $properties));
   if (!empty($element['#value'])) {
     if (is_array($element['#value'])) {
@@ -584,6 +596,11 @@ function theme_select_as_links($vars) {
       $output .= '<input type="hidden" name="' . $name . '" value="' . $element['#value'] . '" />';
     }
   }
+
+  if (isset($element['#bef_enable_fastlivefilter']) && $element['#bef_enable_fastlivefilter']) {
+    $output .= '</div>';
+  }
+
   $output .= '</div>';
 
   return $output;
diff --git a/drupal/sites/all/modules/contrib/better_exposed_filters/better_exposed_filters_exposed_form_plugin.inc b/drupal/sites/all/modules/contrib/better_exposed_filters/better_exposed_filters_exposed_form_plugin.inc
index c3cb8f4..1183ba6 100644
--- a/drupal/sites/all/modules/contrib/better_exposed_filters/better_exposed_filters_exposed_form_plugin.inc
+++ b/drupal/sites/all/modules/contrib/better_exposed_filters/better_exposed_filters_exposed_form_plugin.inc
@@ -511,6 +511,19 @@ Title Desc|Z -> A</pre> Leave the replacement value blank to remove an option al
         '#description' => t('Leave blank to use views default value. <em>Note: overriding this label will break translations and localizations. Leave this field blank if preserving internationalization is important to your site.</em>'),
       );
 
+      // Add jquery-fastLiveFilter plugin
+      $bef_options[$label]['more_options']['enable_fastlivefilter'] = array(
+        '#type' => 'checkbox',
+        '#title' => t('Enable fastLiveFilter plugin for links.'),
+        '#default_value' => $existing[$label]['more_options']['enable_fastlivefilter'],
+        '#states' => array(
+          'visible' => array(
+            ':input[name="enable_fastlivefilter"]' => array('checked' => TRUE),
+          ),
+        ),
+        '#description' => t('Enable fastLiveFilter plugin for links.'),
+      );
+
       // Build a description option form element -- available to all exposed
       // filters.
       $bef_options[$label]['more_options']['bef_filter_description'] = array(
@@ -1173,6 +1186,7 @@ Title Desc|Z -> A</pre> Leave the replacement value blank to remove an option al
         case 'bef_links':
           $show_apply = TRUE;
           $form[$filter_id]['#theme'] = 'select_as_links';
+          $form[$filter_id]['#bef_enable_fastlivefilter'] = $options['more_options']['enable_fastlivefilter'];
 
           // Exposed form displayed as blocks can appear on pages other than
           // the view results appear on. This can cause problems with
@@ -1572,6 +1586,7 @@ Title Desc|Z -> A</pre> Leave the replacement value blank to remove an option al
         'bef_select_all_none_nested' => FALSE,
         'bef_collapsible' => FALSE,
         'is_secondary' => FALSE,
+        'enable_fastlivefilter' => FALSE,
         'bef_filter_description' => '',
         'any_label' => '',
         'tokens' => array(
