diff --git a/drupal/misc/ajax.js b/drupal/misc/ajax.js
index 90c3bb8..06f3df4 100644
--- a/drupal/misc/ajax.js
+++ b/drupal/misc/ajax.js
@@ -448,7 +448,8 @@ Drupal.ajax.prototype.getEffect = function (response) {
  * Handler for the form redirection error.
  */
 Drupal.ajax.prototype.error = function (response, uri) {
-  alert(Drupal.ajaxError(response, uri));
+  // alert(Drupal.ajaxError(response, uri));
+  console.log(Drupal.ajaxError(response, uri));
   // Remove the progress element.
   if (this.progress.element) {
     $(this.progress.element).remove();
