diff --git a/drupal/modules/user/user.module b/drupal/modules/user/user.module
index bdfd36f..d03629d 100644
--- a/drupal/modules/user/user.module
+++ b/drupal/modules/user/user.module
@@ -3123,15 +3123,17 @@ function user_role_grant_permissions($rid, array $permissions = array()) {
   $modules = user_permission_get_modules();
   // Grant new permissions for the role.
   foreach ($permissions as $name) {
-    db_merge('role_permission')
-      ->key(array(
-        'rid' => $rid,
-        'permission' => $name,
-      ))
-      ->fields(array(
-        'module' => $modules[$name],
-      ))
-      ->execute();
+    if (isset($modules[$name])) {
+      db_merge('role_permission')
+        ->key(array(
+          'rid' => $rid,
+          'permission' => $name,
+        ))
+        ->fields(array(
+          'module' => $modules[$name],
+        ))
+        ->execute();
+    }
   }
 
   // Clear the user access cache.
