diff --git a/drupal/sites/all/modules/contrib/ctools/js/modal.js b/drupal/sites/all/modules/contrib/ctools/js/modal.js
index c757ef2..d692e71 100644
--- a/drupal/sites/all/modules/contrib/ctools/js/modal.js
+++ b/drupal/sites/all/modules/contrib/ctools/js/modal.js
@@ -28,6 +28,10 @@
     else if (choice) {
       $.extend(true, opts, choice);
     }
+    // Default modal.
+    else {
+      choice = 'default';
+    }
 
     var defaults = {
       modalTheme: 'CToolsModalDialog',
