diff --git a/drupal/sites/all/modules/contrib/entity_menu_links/entity_menu_links.module b/drupal/sites/all/modules/contrib/entity_menu_links/entity_menu_links.module
index 5379ef0..c4e2a14 100644
--- a/drupal/sites/all/modules/contrib/entity_menu_links/entity_menu_links.module
+++ b/drupal/sites/all/modules/contrib/entity_menu_links/entity_menu_links.module
@@ -81,7 +81,9 @@ function entity_menu_links_menu_link_alter(&$menu_link) {
  */
 function entity_menu_links_menu_link_insert($menu_link) {
   if ($menu_link['module'] == 'menu') {
-    $menu_link['uuid'] = uuid_generate();
+    if (empty($menu_link['uuid'])) {
+      $menu_link['uuid'] = uuid_generate();
+    }
     $menu_link['vid'] = entity_menu_links_insert_revision($menu_link);
 
     db_update('menu_links')
