diff --git a/drupal/sites/all/modules/contrib/features/includes/features.menu.inc b/drupal/sites/all/modules/contrib/features/includes/features.menu.inc
index 1ea7da6..db1653d 100644
--- a/drupal/sites/all/modules/contrib/features/includes/features.menu.inc
+++ b/drupal/sites/all/modules/contrib/features/includes/features.menu.inc
@@ -408,7 +408,7 @@ function features_menu_link_load($identifier) {
  * Returns a lowercase clean string with only letters, numbers and dashes
  */
 function features_clean_title($str) {
-  return strtolower(preg_replace_callback('/(\s)|([^a-zA-Z\-0-9])/i', create_function(
+  return strtolower(preg_replace_callback('/(\p{Z})|([^\w-]+)/u', create_function(
           '$matches',
           'return $matches[1]?"-":"";'
       ), $str));
