diff --git a/drupal/sites/all/modules/contrib/features/includes/features.menu.inc b/drupal/sites/all/modules/contrib/features/includes/features.menu.inc
index a11350f..1ea7da6 100644
--- a/drupal/sites/all/modules/contrib/features/includes/features.menu.inc
+++ b/drupal/sites/all/modules/contrib/features/includes/features.menu.inc
@@ -359,6 +359,7 @@ function features_menu_link_load($identifier) {
      ->fields('menu_links', array('menu_name', 'mlid', 'plid', 'link_path', 'router_path', 'link_title', 'options', 'module', 'hidden', 'external', 'has_children', 'expanded', 'weight', 'customized'))
      ->condition('menu_name', $menu_name)
      ->condition('link_path', $link_path)
+     ->addTag('features_menu_link')
      ->execute()
     ->fetchAllAssoc('mlid');
 
