diff --git a/drupal/sites/all/modules/contrib/features/includes/features.menu.inc b/drupal/sites/all/modules/contrib/features/includes/features.menu.inc
index 3eac19d..0c82c93 100644
--- a/drupal/sites/all/modules/contrib/features/includes/features.menu.inc
+++ b/drupal/sites/all/modules/contrib/features/includes/features.menu.inc
@@ -247,6 +247,9 @@ function menu_links_features_export_render($module, $data, $export = NULL) {
        unset($link['plid']);
        unset($link['mlid']);
 
+      // Sort the link keys to avoid confusing the diff algorithm.
+      ksort($link);
+
       $code[] = "  // Exported menu link: {$new_identifier}";
       $code[] = "  \$menu_links['{$new_identifier}'] = ". features_var_export($link, '  ') .";";
       $translatables[] = $link['link_title'];
