From 7d6ab204b45f4b011fad98ffae837e8ef2496954 Mon Sep 17 00:00:00 2001
From: Caio SBA <caiosba@gmail.com>
Date: Tue, 25 Dec 2012 12:51:56 -0200
Subject: [PATCH] Fix fatal error when type options are not set

---
 includes/flag/flag_flag.inc |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/includes/flag/flag_flag.inc b/includes/flag/flag_flag.inc
index 26ef707..03eb79b 100644
--- a/includes/flag/flag_flag.inc
+++ b/includes/flag/flag_flag.inc
@@ -115,7 +115,7 @@ class flag_flag {
 
     // Populate the options with the defaults.
     $options = (array) unserialize($row->options);
-    $options += $flag->options();
+    $options += (array) $flag->options();
 
     // Make the unserialized options accessible as normal properties.
     foreach ($options as $option => $value) {
@@ -184,7 +184,7 @@ class flag_flag {
 
     // Merge in options from the current link type.
     $link_type = $this->get_link_type();
-    $options = array_merge($options, $link_type['options']);
+    if (isset($link_type) && isset($link_type['options'])) $options = array_merge($options, $link_type['options']);
 
     // Allow other modules to change the flag options.
     drupal_alter('flag_options', $options, $this);
-- 
1.7.2.5

