diff --git a/drupal/sites/all/modules/contrib/flag/theme/flag.js b/drupal/sites/all/modules/contrib/flag/theme/flag.js
index 3f1aedc..82e40e6 100644
--- a/drupal/sites/all/modules/contrib/flag/theme/flag.js
+++ b/drupal/sites/all/modules/contrib/flag/theme/flag.js
@@ -81,7 +81,12 @@ Drupal.flagLink = function(context) {
 
         // Find all the link wrappers on the page for this flag, but exclude
         // the triggering element because Flag's own javascript updates it.
-        var $wrappers = $('.flag-wrapper.flag-' + data.flagName.flagNameToCSS() + '-' + data.contentId).not(data.link);
+        if (typeof data.link != 'undefined') {
+          var $wrappers = $('.flag-wrapper.flag-' + data.flagName.flagNameToCSS() + '-' + data.contentId).not(data.link);
+        }
+        else {
+          var $wrappers = $('.flag-wrapper.flag-' + data.flagName.flagNameToCSS() + '-' + data.contentId);
+        }
         var $newLink = $(data.newLink);
 
         // Hide message, because we want the message to be shown on the triggering element alone.
