diff --git a/drupal/sites/all/modules/contrib/flag/includes/flag/flag_flag.inc b/drupal/sites/all/modules/contrib/flag/includes/flag/flag_flag.inc
index 071b148..dfd29e2 100644
--- a/drupal/sites/all/modules/contrib/flag/includes/flag/flag_flag.inc
+++ b/drupal/sites/all/modules/contrib/flag/includes/flag/flag_flag.inc
@@ -127,7 +127,7 @@ class flag_flag {
 
     // Populate the options with the defaults.
     $options = (array) unserialize($row->options);
-    $options += $flag->options();
+    $options += (array) $flag->options();
 
     // Make the unserialized options accessible as normal properties.
     foreach ($options as $option => $value) {
@@ -196,7 +196,7 @@ class flag_flag {
 
     // Merge in options from the current link type.
     $link_type = $this->get_link_type();
-    $options = array_merge($options, $link_type['options']);
+    if (isset($link_type) && isset($link_type['options'])) $options = array_merge($options, $link_type['options']);
 
     // Allow other modules to change the flag options.
     drupal_alter('flag_options', $options, $this);
