diff --git a/drupal/sites/all/modules/contrib/l10n_update/l10n_update.module b/drupal/sites/all/modules/contrib/l10n_update/l10n_update.module
index 0ec2a8c..8a1dd62 100644
--- a/drupal/sites/all/modules/contrib/l10n_update/l10n_update.module
+++ b/drupal/sites/all/modules/contrib/l10n_update/l10n_update.module
@@ -193,14 +193,12 @@ function l10n_update_locale_translate_edit_form_submit($form, &$form_state) {
   module_load_include('inc', 'l10n_update');
   $lid = $form_state['values']['lid'];
   foreach ($form_state['values']['translations'] as $langcode => $value) {
-    if (!empty($value) && $value != $form_state['complete form']['translations'][$langcode]['#default_value']) {
-      // An update has been made, mark the string as customized.
-      db_update('locales_target')
-        ->fields(array('l10n_status' => L10N_UPDATE_STRING_CUSTOM))
-        ->condition('lid', $lid)
-        ->condition('language', $langcode)
-        ->execute();
-    }
+    // An update has been made, mark the string as customized.
+    db_update('locales_target')
+      ->fields(array('l10n_status' => L10N_UPDATE_STRING_CUSTOM))
+      ->condition('lid', $lid)
+      ->condition('language', $langcode)
+      ->execute();
   }
 }
 
