From 2e9fc1cbd630ecd09afe5611fe9a162d1c1c5c63 Mon Sep 17 00:00:00 2001
From: Caio SBA <caiosba@gmail.com>
Date: Fri, 23 Aug 2013 02:57:54 -0300
Subject: [PATCH] Including library

---
 recaptcha.module |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/recaptcha.module b/recaptcha.module
index d6cb122..560d92e 100644
--- a/recaptcha.module
+++ b/recaptcha.module
@@ -5,6 +5,8 @@
  * Uses the reCAPTCHA web service to improve the CAPTCHA system.
  */
 
+require_once(drupal_get_path('module', 'recaptcha') . '/recaptcha-php-1.11/recaptchalib.php');
+
 /**
  * Implements hook_help().
  */
-- 
1.7.2.5

