diff --git a/drupal/sites/all/modules/contrib/rules/includes/rules.state.inc b/drupal/sites/all/modules/contrib/rules/includes/rules.state.inc
index a4150e7..7166d9e 100644
--- a/drupal/sites/all/modules/contrib/rules/includes/rules.state.inc
+++ b/drupal/sites/all/modules/contrib/rules/includes/rules.state.inc
@@ -616,7 +616,7 @@ class RulesData  {
    * @see RulesPlugin::availableVariables()
    */
   public static function addSiteMetadata(EntityMetadataWrapper $wrapper, $property_info) {
-    $site_info = entity_get_property_info('site');
+    $site_info = entity_get_property_info('site') + array('properties' => array());
     $property_info['properties'] += $site_info['properties'];
     // Also invoke the usual callback for altering metadata, in case actions
     // have specified further metadata.
