diff --git a/drupal/sites/all/modules/contrib/rules/modules/system.eval.inc b/drupal/sites/all/modules/contrib/rules/modules/system.eval.inc
index 0de887c..cc5125f 100644
--- a/drupal/sites/all/modules/contrib/rules/modules/system.eval.inc
+++ b/drupal/sites/all/modules/contrib/rules/modules/system.eval.inc
@@ -190,8 +190,14 @@ class RulesTokenEvaluator extends RulesDataInputEvaluator {
     foreach (token_scan($whole_text) as $var_name => $tokens) {
       $var_name = str_replace('-', '_', $var_name);
       if (isset($var_info[$var_name]) && ($token_type = _rules_system_token_map_type($var_info[$var_name]['type']))) {
-        // We have to key $data with the type token uses for the variable.
-        $data = rules_unwrap_data(array($token_type => $state->get($var_name)), array($token_type => $var_info[$var_name]));
+        if (($entity = $state->get($var_name)) && $entity instanceof EntityDrupalWrapper) {
+          $token_type = $entity->type();
+          $data = rules_unwrap_data(array($token_type => $entity->value()), array($token_type => $var_info[$var_name]));
+        }
+        else {
+          // We have to key $data with the type token uses for the variable.
+          $data = rules_unwrap_data(array($token_type => $state->get($var_name)), array($token_type => $var_info[$var_name]));
+        }
         $replacements += token_generate($token_type, $tokens, $data, $options);
       }
       else {
