From 2679ee95f299e415d856533feff719e3d83da988 Mon Sep 17 00:00:00 2001
From: Caio SBA <caiosba@gmail.com>
Date: Mon, 25 Mar 2013 14:57:56 -0300
Subject: [PATCH] Redirect to user/register

---
 twitter_signin/twitter_signin.module |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/twitter_signin/twitter_signin.module b/twitter_signin/twitter_signin.module
index dfea8ef..2783fa8 100644
--- a/twitter_signin/twitter_signin.module
+++ b/twitter_signin/twitter_signin.module
@@ -183,7 +183,7 @@ function twitter_signin_oauth_callback_submit(&$form, &$form_state) {
 
   if (!$success) {
     $_SESSION['twitter']['values'] = $response;
-    drupal_goto('user/login');
+    drupal_goto('user/register');
   }
   else {
     user_login_finalize();
-- 
1.7.2.5

