diff --git a/drupal/sites/all/modules/contrib/twitter/twitter_signin/twitter_signin.module b/drupal/sites/all/modules/contrib/twitter/twitter_signin/twitter_signin.module
index e44308e..e4d8ad0 100644
--- a/drupal/sites/all/modules/contrib/twitter/twitter_signin/twitter_signin.module
+++ b/drupal/sites/all/modules/contrib/twitter/twitter_signin/twitter_signin.module
@@ -248,7 +248,7 @@ function twitter_signin_oauth_callback_submit(&$form, &$form_state) {
   // If there was a problem, return to the login screen.
   if (!$success) {
     $_SESSION['twitter']['values'] = $response;
-    drupal_goto('user/login');
+    drupal_goto('user/register');
   }
 
   // The visitor successfully logged in.
