diff --git a/drupal/sites/all/modules/contrib/varnish/varnish.install b/drupal/sites/all/modules/contrib/varnish/varnish.install
index c5b9c2d..defface 100644
--- a/drupal/sites/all/modules/contrib/varnish/varnish.install
+++ b/drupal/sites/all/modules/contrib/varnish/varnish.install
@@ -35,6 +35,6 @@ function varnish_requirements($phase) {
         }
       }
     }
-    return $requirements;
   }
+  return $requirements;
 }
