diff --git a/drupal/sites/all/modules/contrib/varnish/varnish.admin.inc b/drupal/sites/all/modules/contrib/varnish/varnish.admin.inc
index d47c142..181459b 100644
--- a/drupal/sites/all/modules/contrib/varnish/varnish.admin.inc
+++ b/drupal/sites/all/modules/contrib/varnish/varnish.admin.inc
@@ -66,6 +66,13 @@ function varnish_admin_settings_form() {
     '#required' => variable_get('varnish_version', 2.1) == '4' ? TRUE : FALSE,
   );
 
+  $form['varnish_control_key_appendnewline'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Append Newline (\n)'),
+    '#default_value' => variable_get('varnish_control_key_appendnewline', TRUE),
+    '#description' => t('Most control keys have an embedded newline at the end. Uncheck this if you know it doesn\'t or if you are reading the file in direct with an fopen.'),
+  );
+
   $form['varnish_socket_timeout'] = array(
     '#type' => 'textfield',
     '#title' => t('Varnish connection timeout (milliseconds)'),
diff --git a/drupal/sites/all/modules/contrib/varnish/varnish.module b/drupal/sites/all/modules/contrib/varnish/varnish.module
index 8342e0c..7c23071 100644
--- a/drupal/sites/all/modules/contrib/varnish/varnish.module
+++ b/drupal/sites/all/modules/contrib/varnish/varnish.module
@@ -345,7 +345,12 @@ function _varnish_terminal_run($commands) {
       if ($status['code'] == 107) {
         $secret = variable_get('varnish_control_key', '');
         $challenge = substr($status['msg'], 0, 32);
-        $pack = $challenge . "\x0A" . $secret . "\x0A" . $challenge . "\x0A";
+        if (variable_get('varnish_control_key_appendnewline',TRUE)) {
+          $pack = $challenge . "\x0A" . $secret . "\x0A" . $challenge . "\x0A";
+        }
+        else {
+          $pack = $challenge . "\x0A". $secret . $challenge . "\x0A";
+        }
         $key = hash('sha256', $pack);
         socket_write($client, "auth $key\n");
         $status = _varnish_read_socket($client);
