diff --git a/drupal/sites/all/modules/contrib/views/handlers/views_handler_field_date.inc b/drupal/sites/all/modules/contrib/views/handlers/views_handler_field_date.inc
index f2c6371..1f7365e 100644
--- a/drupal/sites/all/modules/contrib/views/handlers/views_handler_field_date.inc
+++ b/drupal/sites/all/modules/contrib/views/handlers/views_handler_field_date.inc
@@ -141,7 +141,7 @@ class views_handler_field_date extends views_handler_field {
           if ($custom_format == 'r') {
             return format_date($value, $format, $custom_format, $timezone, 'en');
           }
-          return format_date($value, $format, $custom_format, $timezone);
+          return format_date($value, $format, t($custom_format), $timezone);
         default:
           return format_date($value, $format, '', $timezone);
       }
