diff --git a/drupal/sites/all/modules/contrib/views/plugins/views_plugin_exposed_form.inc b/drupal/sites/all/modules/contrib/views/plugins/views_plugin_exposed_form.inc
index 2e0ad7d..343eee8 100644
--- a/drupal/sites/all/modules/contrib/views/plugins/views_plugin_exposed_form.inc
+++ b/drupal/sites/all/modules/contrib/views/plugins/views_plugin_exposed_form.inc
@@ -324,6 +324,7 @@ class views_plugin_exposed_form extends views_plugin {
       $this->view->exposed_data = array();
     }
 
+    $form_state['redirect'] = current_path();
     $form_state['values'] = array();
   }
 }
