From a3a3cd8c8e0766236d334806a5b10f9a3434bd39 Mon Sep 17 00:00:00 2001
From: Caio SBA <caiosba@gmail.com>
Date: Sun, 17 Feb 2013 23:32:03 -0300
Subject: [PATCH 2/3] Nested views support

---
 views_autorefresh/views_autorefresh.js |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/views_autorefresh/views_autorefresh.js b/views_autorefresh/views_autorefresh.js
index 3bc6afe..4995199 100644
--- a/views_autorefresh/views_autorefresh.js
+++ b/views_autorefresh/views_autorefresh.js
@@ -140,7 +140,7 @@ Drupal.ajax.prototype.commands.viewsAutoRefreshIncremental = function (ajax, res
     var $source = $(response.data).find(sourceSelector).not(sourceSelector + ' ' + sourceSelector).children();
     if ($source.size() > 0) {
       var targetSelector = Drupal.settings.views_autorefresh[response.view_name].incremental.targetSelector || '.view-content';
-      var $target = $view.find(targetSelector);
+      var $target = $view.find(targetSelector).not(targetSelector + ' ' + targetSelector);
 
       // If initial view was empty, remove the empty divs then add the target div.
       if ($target.size() == 0) {
@@ -153,14 +153,14 @@ Drupal.ajax.prototype.commands.viewsAutoRefreshIncremental = function (ajax, res
         }
         else if ($(afterSelector, $view).size() > 0) {
           // insert content after given div.
-          $(afterSelector, $view).after(targetStructure);
+          $view.find(afterSelector).not(targetSelector + ' ' + afterSelector).after(targetStructure);
         }
         else {
           // insert content as first child of view div.
           $view.prepend(targetStructure);
         }
         // Now that it's inserted, find it for manipulation.
-        $target = $view.find(targetSelector);
+        $target = $view.find(targetSelector).not(targetSelector + ' ' + targetSelector);
       }
 
       // Remove first, last row classes from items.
-- 
1.7.2.5

