From ad2c0ab30144aa3d28ce861bd8545b15d053e1a1 Mon Sep 17 00:00:00 2001
From: Caio SBA <caiosba@gmail.com>
Date: Sun, 13 Jan 2013 22:31:59 -0200
Subject: [PATCH] Fixing bug on incremental auto refresh when page changes

---
 views_autorefresh/views_autorefresh.js |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/views_autorefresh/views_autorefresh.js b/views_autorefresh/views_autorefresh.js
index d969756..ad03249 100644
--- a/views_autorefresh/views_autorefresh.js
+++ b/views_autorefresh/views_autorefresh.js
@@ -68,6 +68,7 @@ Drupal.behaviors.views_autorefresh = {
                 Drupal.settings.views_autorefresh[settings.view_name].ajax = new Drupal.ajax(view, this, element_settings);
 
                 // Activate refresh timer.
+                clearTimeout(Drupal.settings.views_autorefresh[settings.view_name].timer);
                 Drupal.views_autorefresh.timer(settings.view_name, anchor, target);
               }); // .each function () {
         }); // $view.filter().each
-- 
1.7.2.5

