From 00a30976da93301d3f72ad452e8c0030d4073fe2 Mon Sep 17 00:00:00 2001
From: Caio SBA <caiosba@gmail.com>
Date: Sun, 17 Feb 2013 23:33:09 -0300
Subject: [PATCH 3/3] Attach behaviors after incrementally adding content

---
 views_autorefresh/views_autorefresh.js |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/views_autorefresh/views_autorefresh.js b/views_autorefresh/views_autorefresh.js
index 4995199..294c6e3 100644
--- a/views_autorefresh/views_autorefresh.js
+++ b/views_autorefresh/views_autorefresh.js
@@ -194,6 +194,9 @@ Drupal.ajax.prototype.commands.viewsAutoRefreshIncremental = function (ajax, res
 
     // Reactivate refresh timer.
     Drupal.views_autorefresh.timer(response.view_name, $('.auto-refresh a', $view), $view);
+    
+    // Attach behaviors
+    Drupal.attachBehaviors($view);
   }
 }
 
-- 
1.7.2.5

