diff --git a/drupal/sites/all/modules/contrib/views_load_more/views_load_more.module b/drupal/sites/all/modules/contrib/views_load_more/views_load_more.module
index 00403c0..a1f9e97 100644
--- a/drupal/sites/all/modules/contrib/views_load_more/views_load_more.module
+++ b/drupal/sites/all/modules/contrib/views_load_more/views_load_more.module
@@ -49,7 +49,7 @@ function views_load_more_views_ajax_data_alter(&$commands, $view) {
       }
       // the replace should the only one, but just in case, we'll make sure.
       if ($command['command'] == 'insert' && $command['selector'] == '.view-dom-id-' . $view->dom_id) {
-        if ($view->style_plugin->options['type'] == 'ul' || $view->style_plugin->options['type'] == 'ol') {
+        if (isset($view->style_plugin->options['type']) && (($view->style_plugin->options['type'] == 'ul') || $view->style_plugin->options['type'] == 'ol')) {
           $target = ".{$view->style_plugin->options['wrapper_class']} > {$view->style_plugin->options['type']}:not(.links)";
           $commands[$key]['targetList'] = $target;
         }
