diff --git a/drupal/sites/all/modules/contrib/views_autorefresh/views/views_autorefresh.views.inc b/drupal/sites/all/modules/contrib/views_autorefresh/views/views_autorefresh.views.inc
index f9f670f..e5959b7 100644
--- a/drupal/sites/all/modules/contrib/views_autorefresh/views/views_autorefresh.views.inc
+++ b/drupal/sites/all/modules/contrib/views_autorefresh/views/views_autorefresh.views.inc
@@ -20,7 +20,12 @@ function views_autorefresh_views_data_alter(&$data) {
         $date = $field_info;
         $date['title'] = $field_info['title'] . ' (with operator)';
         $date['argument']['handler'] = 'views_autorefresh_handler_argument_date';
-        $date['argument']['field'] = empty($date['field']['field']) ? $field_name : $date['field']['field'];
+        if (isset($date['field']['field'])) {
+          $date['argument']['field'] = $date['field']['field'];
+        }
+        else {
+          $date['argument']['field'] = empty($date['real field']) ? $field_name : $date['real field'];
+        }
         unset($date['filter'], $date['field'], $date['relationship'], $date['sort']);
         $data[$table_name][$field_name . '_operator'] = $date;
       }
