diff --git a/drupal/sites/all/modules/contrib/views_load_more/views_load_more.js b/drupal/sites/all/modules/contrib/views_load_more/views_load_more.js
index fadb876..570f91a 100644
--- a/drupal/sites/all/modules/contrib/views_load_more/views_load_more.js
+++ b/drupal/sites/all/modules/contrib/views_load_more/views_load_more.js
@@ -58,6 +58,9 @@
     pager_query = pager_query + ':not(' + wrapper_selector + ' ' + content_query + ' ' + pager_query + ')';
     content_query = '.view-content:not(' + wrapper_selector + ' .view-content .view-content)';
 
+    // Additional processing over new content
+    wrapper.trigger('views_load_more.new_content', new_content.clone());
+
     // If we're using any effects. Hide the new content before adding it to the DOM.
     if (effect.showEffect != 'show') {
       new_content.find(content_query).children().hide();
