diff --git a/includes/database/mysql/database.inc b/includes/database/mysql/database.inc
index 0b84f27..2fa7706 100644
--- a/includes/database/mysql/database.inc
+++ b/includes/database/mysql/database.inc
@@ -27,6 +27,10 @@ class DatabaseConnection_mysql extends DatabaseConnection {
     $this->transactionalDDLSupport = FALSE;
 
     $this->connectionOptions = $connection_options;
+    
+    // Allow the drupal default charset to be overridden in the settings.php 
+    // database config
+    $charset = (array_key_exists('charset', $connection_options) ? $connection_options['charset'] : 'utf8');
 
     // The DSN should use either a socket or a host/port.
     if (isset($connection_options['unix_socket'])) {
@@ -39,7 +43,7 @@ class DatabaseConnection_mysql extends DatabaseConnection {
     // Character set is added to dsn to ensure PDO uses the proper character
     // set when escaping. This has security implications. See
     // https://www.drupal.org/node/1201452 for further discussion.
-    $dsn .= ';charset=utf8';
+    $dsn .= ';charset=' . $charset;
     $dsn .= ';dbname=' . $connection_options['database'];
     // Allow PDO options to be overridden.
     $connection_options += array(
@@ -58,10 +62,10 @@ class DatabaseConnection_mysql extends DatabaseConnection {
     // certain one has been set; otherwise, MySQL defaults to 'utf8_general_ci'
     // for UTF-8.
     if (!empty($connection_options['collation'])) {
-      $this->exec('SET NAMES utf8 COLLATE ' . $connection_options['collation']);
+      $this->exec('SET NAMES ' . $charset . ' COLLATE ' . $connection_options['collation']);
     }
     else {
-      $this->exec('SET NAMES utf8');
+      $this->exec('SET NAMES ' . $charset);
     }
 
     // Set MySQL init_commands if not already defined.  Default Drupal's MySQL
diff --git a/includes/database/mysql/schema.inc b/includes/database/mysql/schema.inc
index 2a2722e..1cb6d7b 100644
--- a/includes/database/mysql/schema.inc
+++ b/includes/database/mysql/schema.inc
@@ -77,11 +77,11 @@ class DatabaseSchema_mysql extends DatabaseSchema {
    */
   protected function createTableSql($name, $table) {
     $info = $this->connection->getConnectionOptions();
-
     // Provide defaults if needed.
     $table += array(
       'mysql_engine' => 'InnoDB',
-      'mysql_character_set' => 'utf8',
+      // allow charset to be overridden in settings.php
+      'mysql_character_set' => (array_key_exists('charset', $info) ? $info['charset'] : 'utf8'),
     );
 
     $sql = "CREATE TABLE {" . $name . "} (\n";
@@ -129,6 +129,14 @@ class DatabaseSchema_mysql extends DatabaseSchema {
    *   The field specification, as per the schema data structure format.
    */
   protected function createFieldSql($name, $spec) {
+    $info = $this->connection->getConnectionOptions();
+    if (isset($spec['length']) && isset($info['charset'])) {
+      // If the length of the field is set and the collation is utf8mb4
+      // length should be 191 and not 255
+      if ($spec['length'] >= 192 && $info['charset'] == "utf8mb4") {
+        $spec['length'] = 191;
+      }
+    }
     $sql = "`" . $name . "` " . $spec['mysql_type'];
 
     if (in_array($spec['mysql_type'], array('VARCHAR', 'CHAR', 'TINYTEXT', 'MEDIUMTEXT', 'LONGTEXT', 'TEXT'))) {
